src/main/java/com/google/api/generator/spring/composer/Utils.java

```diff
@@ -0,0 +1,176 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.spring.composer;
+
+import com.google.api.generator.engine.ast.AstNode;
+import com.google.api.generator.engine.ast.Expr;
+import com.google.api.generator.engine.ast.MethodInvocationExpr;
+import com.google.api.generator.engine.ast.PrimitiveValue;
+import com.google.api.generator.engine.ast.TypeNode;
+import com.google.api.generator.engine.ast.ValueExpr;
+import com.google.api.generator.gapic.composer.store.TypeStore;
+import com.google.api.generator.gapic.model.GapicContext;
+import com.google.api.generator.gapic.model.GapicRetrySettings;
+import com.google.api.generator.gapic.model.GapicServiceConfig;
+import com.google.api.generator.gapic.model.Method;
+import com.google.api.generator.gapic.model.Service;
+import com.google.common.base.CaseFormat;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.protobuf.Duration;
+import com.google.protobuf.util.Durations;
+import io.grpc.serviceconfig.MethodConfig.RetryPolicy;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+public class Utils {
+  private static final TypeStore FIXED_TYPESTORE = createStaticTypes();
+
+  private static final String BRAND_NAME = "spring.cloud.gcp";
+
+  public static String getLibName(GapicContext context) {
+    String pakkageName = context.services().get(0).pakkage();
+    List<String> pakkagePhrases = Splitter.on(".").splitToList(pakkageName);
+    // TODO: confirm if this is guaranteed pattern: xx.[...].xx.lib-name.v[version]
+    // eg. for vision proto: "com.google.cloud.vision.v1"
+    // https://github.com/googleapis/java-vision/blob/main/proto-google-cloud-vision-v1/src/main/proto/google/cloud/vision/v1/image_annotator.proto#L36
+    return pakkagePhrases.get(pakkagePhrases.size() - 2);
+  }
```

Comment:
Is there a better way to get client library name? OR is service name unique across libraries?
Where this is used:
 - prefix for Spring properties: e.g. [prefix].[libraryName].[serviceName].enabled
 - descriptive content in `SpringWriter.writeSpringAdditionalMetadataJson()`, might also needed in other descriptive content e.ge.package.info/documentation.
   - For this purpose: use title as in [package-info.java](https://github.com/googleapis/java-vision/blob/30dd9039609ecb92c45c2f06a85931be5a96ff8f/google-cloud-vision/src/main/java/com/google/cloud/vision/v1/package-info.java#L18), which is obtained from [context.serviceYamlProto().getTitle()](https://github.com/googleapis/gapic-generator-java/blob/f8b36168372118d6ed639263bae192647912ed43/src/main/java/com/google/api/generator/gapic/composer/ClientLibraryPackageInfoComposer.java#L68-L73). In case of java-vision, it’s `Cloud Vision API`